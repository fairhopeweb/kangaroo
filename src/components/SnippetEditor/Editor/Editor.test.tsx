import { Provider } from 'react-redux';
import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configureMockStore } from '@jedmao/redux-mock-store';
import renderer from 'react-test-renderer';

import { ThemeType } from 'components/Theme/Theme';
import Editor from './Editor';

jest.mock('ace-builds/webpack-resolver', () => jest.fn());

configure({ adapter: new Adapter() });

describe('<Editor />', () => {
  const mockStore = configureMockStore();
  const mockProps = {
    gutter: true,
    onChange: jest.fn(),
  };
  const initialState = { ui: { theme: ThemeType.DARK, leftPanelWidth: 200 } };

  let store;

  it('render without crashing', () => {
    store = mockStore(initialState);
    shallow(
      <Provider store={store}>
        <Editor {...mockProps} />
      </Provider>,
    );
  });

  it('matches snapshot', () => {
    store = mockStore(initialState);
    const wrapper = renderer.create(
      <Provider store={store}>
        <Editor {...mockProps} />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
