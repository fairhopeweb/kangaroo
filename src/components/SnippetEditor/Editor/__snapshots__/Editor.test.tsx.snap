// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Editor /> matches snapshot 1`] = `
<div
  className="editor"
>
  <div
    id="ace-editor"
    style={
      Object {
        "height": "691px",
        "width": "824px",
      }
    }
  />
</div>
`;
